rsocket-core/src/main/java/io/rsocket/frame/ExtensionFrameFlyweight.java

```diff
@@ -16,13 +17,16 @@ public static ByteBuf encode(
 
     int flags = FrameHeaderFlyweight.FLAGS_I;
 
-    if (metadata != null) {
+    if (metadata != null && metadata != Unpooled.EMPTY_BUFFER) {
```

Comment:
Don't change any of the frame logic with this pull request expect switching to TupleByteBuf - the branch original comes from an older branch and we should defer the the framing logic to the develop branch

Reply:
I'm just wondering whether empty metadata can be counted as "we have metadata, but it is empty"

Reply:
fixed